File: HW/HW1/HW1.py
class Solution(object):
    def palindrome(self, s):     #inputs: string s

        day = ""        #a blank string to get rid of spaces and numbers and the like
        for currentLetter in s: #cycle thru the letters in string
            if (currentLetter.isalpha()):   #if its an actual letter
                day += currentLetter        #add it to our letters-only string

        day = day.lower()         #AND LOWER IT god ur so picky
        oppositeday = day[::-1]   #tbh i just looked up "how to reverse string" and i guess this slices it up all fancy lol. then i save it as opposite day

        if (oppositeday == day) : #if it still equals the og string then yayyyy woohooooo
            return(True)

        return(False)       #else, return false :(

    #ughhhh the cycling thru and making sure all the letters r lowercase is such a painnnnnn


test = Solution()
print("tacocat being a palindrome is", test.palindrome("tacocat"))
print("burritocat being a palindrome is", test.palindrome("burritocat"))
print("Xx_TACoCAT_xX, which sounds like a username I'd see on Scratch, being a palindrome is", test.palindrome("Xx_TACoCAT_xX"))

File: HW/HW1/HW1.java
class Solution {
    
    public static boolean isPalindrome(String s) {
        
        //no isAlpha in java... gonna do it like this
        String day = s.replaceAll("\\p{P}", "");  //gets rid of puncutation
        day = day.replaceAll("0-9", ""); //gets rid of 0-9
        day = day.replaceAll("\\s+",""); //gets rid of spaces?

        String oppositeday = "";        //again start with empty string

        for (int i = day.length() - 1 ; i >= 0 ; i--) { //go backwards starting from end of string
            oppositeday += day.charAt(i);               //add current chara to oppositeday
        }
        //System.out.println(day);
        //System.out.println(oppositeday);
        if (oppositeday.equalsIgnoreCase(day)) {        //can ignore case with this B)
            return true;                                //if oppday = day, return true
        }
        else {
            return false;                               //else, return false
        }
    }
    
    public static void main(String[] args) {
        //tests
        String test1 = "Xx_TaCoCAT_xX";
        System.out.println(isPalindrome(test1));
        
        String test2 = "tacocat";
        System.out.println(isPalindrome(test2));
        
        String test3 = "burritocat";
        System.out.println(isPalindrome(test3));
    }
}
